breaks with a narrow python build

$ python add_svg_glyphs.py
Traceback (most recent call last):
  File "add_svg_glyphs.py", line 37, in <module>
    import add_emoji_gsub
  File "/Users/gilbahat/fonts/noto-fonts-master/add_emoji_gsub.py", line 28, in <module>
    from nototools import font_data
  File "/Users/gilbahat/fonts/noto-fonts-master/nototools/font_data.py", line 21, in <module>
    from nototools import opentype_data
  File "/Users/gilbahat/fonts/noto-fonts-master/nototools/opentype_data.py", line 330, in <module>
    _set_ompl()
  File "/Users/gilbahat/fonts/noto-fonts-master/nototools/opentype_data.py", line 33, in _set_ompl
    unicode_data.load_data()
  File "/Users/gilbahat/fonts/noto-fonts-master/nototools/unicode_data.py", line 75, in load_data
    _load_unicode_data_txt()
  File "/Users/gilbahat/fonts/noto-fonts-master/nototools/unicode_data.py", line 457, in _load_unicode_data_txt
    decomposition = [unichr(int(char, 16)) for char in decomposition]
ValueError: unichr() arg not in range(0x10000) (narrow Python build)

a narrow python build will be a common situation... and specifically there's this: https://github.com/Homebrew/homebrew/pull/32368 :(

still no dice. build also breaks for wide python:
(wide)
Parsing 'GSUB' table...
Traceback (most recent call last):
  File "/usr/local/bin/ttx", line 9, in <module>
    load_entry_point('fonttools==3.0', 'console_scripts', 'ttx')()
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttx.py", line 357, in main
    process(jobs, options)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttx.py", line 336, in process
    action(input, output, options)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttx.py", line 255, in ttCompile
    ttf.save(output)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttLib/__init__.py", line 202, in save
    self._writeTable(tag, writer, done)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttLib/__init__.py", line 631, in _writeTable
    tabledata = self.getTableData(tag)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttLib/__init__.py", line 644, in getTableData
    return self.tables[tag].compile(self)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttLib/tables/_c_m_a_p.py", line 82, in compile
    chunk = table.compile(ttFont)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttLib/tables/_c_m_a_p.py", line 984, in compile
    raise KeyError(name)
KeyError: u'uDEAB'
make: *** [NotoColorEmoji.tmpl.ttf] Error 1
rm NotoColorEmoji.tmpl.ttx

(narrow)
Parsing 'GSUB' table...
Traceback (most recent call last):
  File "/usr/local/bin/ttx", line 9, in <module>
    load_entry_point('fonttools==3.0', 'console_scripts', 'ttx')()
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttx.py", line 357, in main
    process(jobs, options)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttx.py", line 336, in process
    action(input, output, options)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttx.py", line 255, in ttCompile
    ttf.save(output)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttLib/__init__.py", line 202, in save
    self._writeTable(tag, writer, done)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttLib/__init__.py", line 631, in _writeTable
    tabledata = self.getTableData(tag)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttLib/__init__.py", line 644, in getTableData
    return self.tables[tag].compile(self)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttLib/tables/_c_m_a_p.py", line 82, in compile
    chunk = table.compile(ttFont)
  File "/usr/local/lib/python2.7/site-packages/fonttools-3.0-py2.7.egg/fontTools/ttLib/tables/_c_m_a_p.py", line 984, in compile
    raise KeyError(name)
KeyError: u'uDEAB'
make: *** [NotoColorEmoji.tmpl.ttf] Error 1
rm NotoColorEmoji.tmpl.ttx

--- a/nototools/glyph_area_pen.py
+++ b/nototools/glyph_area_pen.py
@@ -12,70 +12,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from fontTools.pens.areaPen import AreaPen
 
-from fontTools.pens.basePen import BasePen
 
+class GlyphAreaPen(AreaPen):
 
-class GlyphAreaPen(BasePen):
-    """Pen used for calculating the area of the contours in a glyph."""
-
-    def __init__(self, glyph_set):
-        super(GlyphAreaPen, self).__init__(glyph_set)
-        self.area = 0
-
-    def unload(self):
-        """Return and then reset the calculated area."""
-
-        area = self.area
-        self.area = 0
-        return area
-
-    def _moveTo(self, pt):
-        """Remember the first point in this contour, in case it's closed. Also
-        set the initial value for p0 in this contour, which will always refer to
-        the most recent point.
-        """
-
-        self.first = pt
-        self.p0 = pt
-
-    def _lineTo(self, pt):
-        """Add the signed area beneath the line from the latest point to this
-        one. Signed areas cancel each other based on the horizontal direction of
-        the line.
-        """
-
-        (x0, y0), (x1, y1) = self.p0, pt
-        self.area += (x1 - x0) * (y1 + y0) / 2.0
-        self.p0 = pt
-
-    def _qCurveToOne(self, p1, p2):
-        """Add the signed area of this quadratic curve.
-        https://github.com/Pomax/bezierinfo/issues/44
-        """
-
-        x0, y0 = self.p0
-        x1, y1 = p1[0] - x0, p1[1] - y0
-        x2, y2 = p2[0] - x0, p2[1] - y0
-        self.area += (x2 * y1 - x1 * y2) / 3.0
-        self.lineTo(p2)
-
-    def _curveToOne(self, p1, p2, p3):
-        """Add the signed area of this cubic curve.
-        https://github.com/Pomax/bezierinfo/issues/44
-        """
-
-        x0, y0 = self.p0
-        x1, y1 = p1[0] - x0, p1[1] - y0
-        x2, y2 = p2[0] - x0, p2[1] - y0
-        x3, y3 = p3[0] - x0, p3[1] - y0
-        self.area += (
-            x1 * (   -     y2 -     y3) +
-            x2 * (y1          - 2 * y3) +
-            x3 * (y1 + 2 * y2         )
-        ) * 3.0 / 20
-        self.lineTo(p3)
-
-    def _closePath(self):
-        """Add the area beneath this contour's closing line."""
-        self.lineTo(self.first)
+    def pop(self):
+        value = self.value
+        self.value = 0
+        return value

--- a/nototools/shape_diff.py
+++ b/nototools/shape_diff.py
@@ -32,7 +32,7 @@ import subprocess
 import tempfile
 
 from fontTools.ttLib import TTFont
-from glyph_area_pen import GlyphAreaPen
+from nototools.glyph_area_pen import GlyphAreaPen
 
 from nototools import hb_input
 
@@ -69,9 +69,9 @@ class ShapeDiffFinder:
         mismatched = {}
         for name in self.names:
             self.glyph_set_a[name].draw(pen_a)
-            area_a = pen_a.unload()
+            area_a = pen_a.pop()
             self.glyph_set_b[name].draw(pen_b)
-            area_b = pen_b.unload()
+            area_b = pen_b.pop()
             if area_a != area_b:
                 mismatched[name] = (area_a, area_b)
 

--- a/nototools/unicode_data.py
+++ b/nototools/unicode_data.py
@@ -32,6 +32,8 @@ import os
 from os import path
 import re
 import sys
+
+from fontTools.misc.py23 import unichr
 try:
   import unicodedata2 as unicodedata  # Unicode 8 compliant native lib
 except ImportError:

--- a/nototools/unittests/font_tests.py
+++ b/nototools/unittests/font_tests.py
@@ -26,7 +26,7 @@ from nototools import coverage
 from nototools import font_data
 from nototools import noto_fonts
 from nototools import unicode_data
-from nototools.pens.glyph_area_pen import GlyphAreaPen
+from nototools.glyph_area_pen import GlyphAreaPen
 from nototools.unittests import layout
 
 
@@ -57,7 +57,7 @@ def load_fonts(patterns, expected_count=None, font_class=None):
         all_font_files += glob.glob(pattern)
     all_fonts = [font_class(font) for font in all_font_files]
     if expected_count:
-        assert len(all_font_files) == expected_count
+        assert len(all_font_files) == expected_count, "got %d fonts, expected %d." % (len(all_font_files), expected_count)
     return all_font_files, all_fonts
 
 
@@ -592,9 +592,9 @@ class TestGlyphAreas(unittest.TestCase):
             if name in self.whitelist:
                 continue
             master_a[name].draw(pen_a)
-            area_a = pen_a.unload()
+            area_a = pen_a.pop()
             master_b[name].draw(pen_b)
-            area_b = pen_b.unload()
+            area_b = pen_b.pop()
             if area_a == area_b:
                 if area_a:
                     self.unchanged.add(name)
@@ -623,7 +623,7 @@ class TestGlyphAreas(unittest.TestCase):
         areas = {}
         for name in self.instance_glyphs_to_test:
             standard[name].draw(pen)
-            areas[name] = pen.unload()
+            areas[name] = pen.pop()
 
         for other in glyph_sets[1:]:
             other_pen = GlyphAreaPen(other)
@@ -631,7 +631,7 @@ class TestGlyphAreas(unittest.TestCase):
                 if name in self.whitelist:
                     continue
                 other[name].draw(other_pen)
-                other_area = other_pen.unload()
+                other_area = other_pen.pop()
                 if name in self.unchanged or not area:
                     self.assertEqual(
                         area, other_area,

